stl/inc/format

```diff
@@ -0,0 +1,363 @@
+// format standard header
+
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#ifndef _FORMAT_
+#define _FORMAT_
+#include <yvals_core.h>
+#if _STL_COMPILER_PREPROCESSOR
+#if !_HAS_CXX20
+#pragma message("The contents of <format> are available only with C++20 or later.")
+#else // ^^^ !_HAS_CXX20 / _HAS_CXX20 vvv
+
+#include <cassert>
+#include <charconv>
+#include <concepts>
+#include <exception>
+#include <iterator>
+#include <locale>
+#include <string>
+#include <string_view>
+#include <variant>
+
+#pragma pack(push, _CRT_PACKING)
+#pragma warning(push, _STL_WARNING_LEVEL)
+#pragma warning(disable : _STL_DISABLED_WARNINGS)
+_STL_DISABLE_CLANG_WARNINGS
+#pragma push_macro("new")
+#undef new
+
+_STD_BEGIN
+
+
+class format_error : public _STD runtime_error {
+    using _STD runtime_error::runtime_error;
+};
+
+enum class _Align { _None, _Left, _Right, _Center };
+
+struct _Auto_id_tag {};
+
+// clang-format off
+template <typename _Ty, typename _CharT>
+concept _Parse_spec_callbacks = requires(_Ty _At, _STD basic_string_view<_CharT> _Sv, _Align _Aln) {
+    { _At._On_align(_Aln) } -> same_as<void>;
+    { _At._On_fill(_Sv) } -> same_as<void>;
+    { _At._On_width(_STD declval<int>()) } -> same_as<void>;

```

Comment:
is specializing declval expensive, I went with it because that way I don't have to look up what some type some identifier is when reading the expression. It is repeated though.